### PR TITLE
35 gather quiz questions

### LINKED ISSUE #35: Gather quiz questions
As a user, I want to see real drivers license questions, in order to crush the test.

**Acceptance Criteria:**
- [x] Find questions we can quiz the user
  - [x] Determine what country we are taking the questions from
- [x] Aim for 10 - 30 questions for each category and 5 - 9 categories
  - [x] Decide question and category count with major stakeholders
- [x] Each question is associated with a source, from which the question is obtained.
- [x] Store these questions in an android database
  - [x] Have a look at room (https://developer.android.com/training/data-storage/room#java)
- [x] Set the quiz state questions to these newly added questions
- [x] Test the implementation manually

### PR DESCRIPTION
**Pull request by JohnWebb00**

### CODE REVIEW

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/QuestionsDAO.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.example.smartcarmqttapp;
+
+import androidx.room.Dao;
+import androidx.room.Query;
+
+import java.util.List;
+
+@Dao
+public interface QuestionsDAO {
```
Comment (GeorgZs) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/QuestionsDAO.java]: @JohnWebb00 Could you maybe explain the exact nature of this class again, the only relevance of this class I see is to query the DB and call method getAll(); returning a list of questions

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/model/Question.java [hunk 2]
```diff
@@ -12,7 +12,7 @@ public class Question {
     private String fourthAnswer;
     private int correctIndex;
     private Category category;
-    private boolean needsReview;
+    private int needsReview;
```
Comment (GeorgZs) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/model/Question.java]: Maybe good to add a short little comment to explain the states that needs review may have: Like what does a 2 mean or any other number for that matter

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/CrushersDataBase.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    public void onUpgrade(SQLiteDatabase db, int pastVersion, int newVersion) {
+    db.execSQL("DROP TABLE IF EXISTS " + QuestionsTable.TABLE_NAME);
+    onCreate(db);
+    }
+
+    private List<Question> createQuestionsAddToList(){
+        //Place holder question values (3 topics(categories), 15 questions each)
+        return Arrays.asList(
+                //CATEGORY 1
+                new Question("placeholder question","answer1","answer2","answer3","answer4", 1, "explanation", 0, "cat1"),
```
Comment (GeorgZs) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/CrushersDataBase.java]: I know they are placeholders but the catergories for all questions are the same... is that intentional given the comments for separating the categories

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/QuestionsDAO.java (hunk 1 above)
Comment (JohnWebb00) [reply SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/QuestionsDAO.java]: This is my mistake. I forgot to remove the class after we scrapped the idea of using the room database. I will delete the class

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/CrushersDataBase.java (hunk 3 above)
Comment (JohnWebb00) [reply SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/CrushersDataBase.java]: Yes, the comments are intended to separate the categories. I can change each question's category field to its respective category and push it again.

**Review by GeorgZs — APPROVED** [review]

**Pull request merged**